pkg/distributor/distributor.go

```diff
@@ -1125,8 +1125,12 @@ func (d *Distributor) send(ctx context.Context, ingester ring.InstanceDesc, time
 		Source:     source,
 	}
 	_, err = c.Push(ctx, &req)
+	if errors.Is(err, context.DeadlineExceeded) {
+		return errors.Wrapf(err, "timed out pushing to ingester (configured timeout: %s)",
+			d.cfg.RemoteTimeout.String())
```

Comment:
Is the `d.cfg.RemoteTimeout` timeout the only possible reason why contexts can get canceled? 
I'm not 100% sure, but I thought there are also other reasons for contexts to get canceled, like for example if the client closes the connection...

Reply:
@replay wouldn't the error for a canceled context be [`context.Canceled`](https://pkg.go.dev/context#pkg-variables)?

Reply:
@replay OTOH, is there any other reason why the context's deadline would pass (and for `context.DeadlineExceeded` to be returned)?

Reply:
Oh, nvm me, I think you're right 👍